src/benchmark_api_internal.cc

```diff
@@ -78,6 +78,11 @@ BenchmarkInstance::BenchmarkInstance(Benchmark* benchmark, int family_idx,
   if (!benchmark_.thread_counts_.empty()) {
     name_.threads = StrFormat("threads:%d", threads_);
   }
+
+#if defined(BENCHMARK_HAS_CXX11)
```

Comment:
How about if it's *used* in cxx03, then it won't be able to use these new `SetUp()/Teardown()`?
Or alternatively, #ifdef here so that it takes a pointer in cxx03 and std::function in cxx11? (not sure this would be better - only more confusing)

Our styleguide (at least internally)  prefers `std::function` over raw function pointer (unless there is legit performance concern, which doesn't seem apply here since ).
Furthermore, how long are we planning on supporting 03? I understand we don't want to break existing users, but limiting our API simply due to compatibility with a (very) old C++ version doesn't seem ideal.

Reply:
What i'm saying is that `Benchmark` class **can not** have different layout (different member variables)
depending on the C++ standard with which the TU into which it's included is compiled,
because the library will itself be compiled with C++11+, and it will expect those fields to be there.
This is a fundamental correctness problem.

Reply:
as an OSS project, there are some parts of the style guide we can't/don't necessarily support.  another example is in non-const reference arguments, which we pioneered ;)

C++03 compatibility is still required, even if support is best effort, due to use on platforms where the standard, although old, is the only one supported by compilers for that platform.

Reply:
Ok, that's reasonable. Let's make these take an function pointers.